### PR TITLE
feat: prometheus translation add support for rw2

### PR DESCRIPTION
**Pull request by jmichalek132**
**Description:** <Describe what has changed.>

Draft starting the work on adding support for remote write 2.0 in the translation package.

**Link to tracking Issue:** <Issue number if applicable> #33661

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: pkg/translator/prometheusremotewrite/helper_v2.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+	ts = &writev2.TimeSeries{}
+	ts.LabelsRefs = c.symbolTable.SymbolizeLabels(lbls, ts.LabelsRefs)
+	c.unique[0] = ts
+	return ts, true
+}
+
+// createAttributes creates a slice of Prometheus Labels with OTLP attributes and pairs of string values.
+// Unpaired string values are ignored. String pairs overwrite OTLP labels if collisions happen and
+// if logOnOverwrite is true, the overwrite is logged. Resulting label names are sanitized.
+func createAttributesV2(resource pcommon.Resource, attributes pcommon.Map, externalLabels map[string]string,
```
Comment (ArthurSens) [inline pkg/translator/prometheusremotewrite/helper_v2.go:57]: I wonder if we could reduce the scope of the first PR even further. Maybe just translate gauges datapoints to samples, without any labels.

File: pkg/translator/prometheusremotewrite/metrics_to_prw_v2_test.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+		ExportCreatedMetric: false,
+		AddMetricSuffixes:   false,
+		SendMetadata:        false,
+	}
+
+	payload := createExportRequest(5, 0, 1, 3, 0)
+
+	tsMap, err := FromMetricsV2(payload.Metrics(), settings)
+	require.NoError(t, err)
+	require.NotNil(t, tsMap)
```
Comment (ArthurSens) [inline pkg/translator/prometheusremotewrite/metrics_to_prw_v2_test.go:25]: You plan to change this, right? Like, asserting that we return real metrics

File: pkg/translator/prometheusremotewrite/number_data_points_v2.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+		pt := dataPoints.At(x)
+		labels := createAttributesV2(
+			resource,
+			pt.Attributes(),
+			settings.ExternalLabels,
+			nil,
+			true,
+			model.MetricNameLabel,
+			name,
+		)
```
Comment (ArthurSens) [inline pkg/translator/prometheusremotewrite/number_data_points_v2.go:28]: What do you think about ignoring labels for the first PR?

**Review by ArthurSens — COMMENTED** [review]
Thank you @jmichalek132! Super nice to see progress here :)

I think we can make this initial task even smaller, so we can get something mergeable faster. What do you think?

**Comment by jmichalek132** [comment]
Closing in favor of #35703.

**Pull request closed without merging**